packages/shorebird_cli/lib/src/commands/release_new/android_release_pipeline.dart

```diff
@@ -0,0 +1,172 @@
+      );
+    }
+
+    final projectRoot = shorebirdEnv.getShorebirdProjectRoot()!;
+    try {
+      return shorebirdAndroidArtifacts.findAab(
+        project: projectRoot,
+        flavor: flavor,
+      );
+    } on MultipleArtifactsFoundException catch (error) {
+      throw ArtifactBuildException(
+        'Build succeeded, but it generated multiple AABs in the '
+        'build directory. ${error.foundArtifacts.map((e) => e.path)}',
+      );
+    } on ArtifactNotFoundException catch (error) {
+      throw ArtifactBuildException(
+        'Build succeeded, but could not find the AAB in the build directory. '
+        'Expected to find ${error.artifactName}',
+      );
+    }
+  }
+
+  @override
+  // TODO: implement postReleaseInstructions
+  String get postReleaseInstructions {
+    final aabFile = shorebirdAndroidArtifacts.findAab(
+      project: projectRoot,
+      flavor: flavor,
+    );
+
+    final String? apkText;
+    if (generateApk) {
+      final apkFile = shorebirdAndroidArtifacts.findApk(
+        project: projectRoot,
+        flavor: flavor,
+      );
+      apkText = generateApk
+          ? '''
+
+Or distribute the apk:
+${lightCyan.wrap(apkFile.path)}
+'''
+          : '';
+    } else {
+      apkText = '';
+    }
+
+    return '''
+
+Your next step is to upload the app bundle to the Play Store:
+${lightCyan.wrap(aabFile.path)}
+$apkText
+For information on uploading to the Play Store, see:
+${link(uri: Uri.parse('https://support.google.com/googleplay/android-developer/answer/9859152?hl=en'))}
+''';
+  }
+
+  @override
+  ReleasePlatform get releasePlatform => ReleasePlatform.android;
+
+  @override
+  Future<void> validatePreconditions() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+        validators: doctor.androidCommandValidators,
+      );
+    } on PreconditionFailedException catch (e) {
+      exit(e.exitCode.code);
+    }
+  }
+
+  @override
+  Future<void> validateArgs() async {
+    if (generateApk && splitApk) {
+      logger
+        ..err(
+          'Shorebird does not support the split-per-abi option at this time',
+        )
+        ..info(
+          '''
+Split APKs are each given a different release version than what is specified in the pubspec.yaml.
+
+See ${link(uri: Uri.parse('https://github.com/flutter/flutter/issues/39817'))} for more information about this issue.
+Please comment and upvote ${link(uri: Uri.parse('https://github.com/shorebirdtech/shorebird/issues/1141'))} if you would like shorebird to support this.''',
+        );
+      throw ExitCode.unavailable.code;
+    }
+  }
+
+  @override
+  Future<String> getReleaseVersion({
+    required FileSystemEntity releaseArtifactRoot,
+  }) async {
+    try {
+      return await artifactManager.extractReleaseVersionFromAppBundle(
+        releaseArtifactRoot.path,
+      );
+    } catch (error) {
+      throw ExitCode.software.code;
```

Comment:
Random thought that came to me

Should we keep a `.shorebird/cli.log` file where we would log crashes. Maybe we could ask for it from users when we be errors on the CLI to help us debug issues.

Reply:
Love it. Filed https://github.com/shorebirdtech/shorebird/issues/2006.